pkg/volume/csi/csi_attacher.go

```diff
@@ -166,6 +168,46 @@ func (c *csiAttacher) waitForVolumeAttachmentInternal(volumeHandle, attachID str
 	return attach.Name, nil
 }
 
+func (c *csiAttacher) waitForVolumeAttachmentWithLister(volumeHandle, attachID string, pollInterval, timeout time.Duration) error {
+	klog.V(4).Info(log("probing VolumeAttachment [id=%v]", attachID))
+
+	ticker := time.NewTicker(pollInterval)
+	timer := time.NewTimer(timeout) // TODO (vladimirvivien) investigate making this configurable
+	defer ticker.Stop()
+	defer timer.Stop()
+
+Loop:
+	for {
+		select {
+		case <-ticker.C:
+			volumeAttachment, err := c.plugin.volumeAttachmentLister.Get(attachID)
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					// Ignore "not found" errors in case the VolumeAttachment was just created and hasn't yet made it into the lister.
```

Comment:
Can we add a log message? I think should be very rare..

Reply:
If we move this check to the top of the loop (outside of the ticker), then this log message seems very likely, because we would start checking the lister immediately after creating the object in the API, and the staleness latency of the lister could be hundreds of milliseconds or more depending on API server load.